### PR TITLE
haskell: add a ghcDefault pkgset, pointing to current default

### PR DESCRIPTION
**Pull request by Profpatsch**
Slightly more semantic.

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5235,7 +5235,7 @@ with pkgs;
 
   haskell = callPackage ./haskell-packages.nix { };
 
-  haskellPackages = haskell.packages.ghc802.override {
+  haskellPackages = haskell.packages.ghcDefault.override {
```
Comment (jb55) [inline pkgs/top-level/all-packages.nix]: any reason not to have just `ghc` instead?

Comment (Profpatsch) [inline pkgs/top-level/all-packages.nix]: Well, the whole point of this is that there was no „default“ packageset before. What you refer to is the compiler derivation (`haskell.compiler.ghc`).

**Comment by peti** [comment]
IMHO, `haskellPackages` is the default package set in Nixpkgs. I don't see what we gain by introducing yet another name for it.

**Comment by Profpatsch** [comment]
Yeah, I suppose we can leave it like that. Often there’s something like `inherit (haskell.packages) ghc802`, so I think it’s nice and consistent to have the default pointer in the same namespace as well, akin to `ghcHEAD`.

**Comment by peti** [comment]
IMHO, people who inherit the ghc802 package set *should* probably inherit `haskellPackages`, really, unless they really want to hard-code a specific version.

**Pull request closed without merging**